CMakeLists.txt

```diff
@@ -1239,20 +1239,11 @@ if(CONFIG_GEN_ISR_TABLES)
   # isr_tables.c is generated from ${ZEPHYR_LINK_STAGE_EXECUTABLE} by
   # gen_isr_tables.py
   add_custom_command(
-    OUTPUT isr_tables.c isrList.bin
-    COMMAND $<TARGET_PROPERTY:bintools,elfconvert_command>
-            $<TARGET_PROPERTY:bintools,elfconvert_flag>
-            $<TARGET_PROPERTY:bintools,elfconvert_flag_intarget>${OUTPUT_FORMAT}
-            $<TARGET_PROPERTY:bintools,elfconvert_flag_outtarget>binary
-            $<TARGET_PROPERTY:bintools,elfconvert_flag_section_only>.intList
-            $<TARGET_PROPERTY:bintools,elfconvert_flag_infile>$<TARGET_FILE:${ZEPHYR_LINK_STAGE_EXECUTABLE}>
-            $<TARGET_PROPERTY:bintools,elfconvert_flag_outfile>isrList.bin
-            $<TARGET_PROPERTY:bintools,elfconvert_flag_final>
+    OUTPUT isr_tables.c
```

Comment:
proposed change fails with armclang:
```
$ ninja -C build 
ninja: Entering directory `build'
[1/134] Preparing syscall dependency handling

[2/134] Generating include/generated/version.h
-- Zephyr version: 3.4.99 (/projects/github/ncs/zephyr), build: zephyr-v3.4.0-4679-g2327fc53e19d
[131/134] Generating isr_tables.c
FAILED: zephyr/isr_tables.c /projects/github/ncs/zephyr/build/zephyr/isr_tables.c 
cd /projects/github/ncs/zephyr/build/zephyr && /usr/bin/python3 /projects/github/ncs/zephyr/scripts/build/gen_isr_tables.py --output-source isr_tables.c --kernel /projects/github/ncs/zephyr/build/zephyr/zephyr_pre0.elf --sw-isr-table --vector-table
gen_isr_tables.py: error: Empty ".intList" section!

ninja: build stopped: subcommand failed.
```

Reply:
The generated linker_zephyr_pre0.cmd file for armclang contains:
```
556 IDT_LIST 0x20040000 NOCOMPRESS 0x800
557 {
558 
  intList 0x20040000
  {
    *.o(.irq_info, +First)
    *.o(.intList)
  }
}
```

The section name is without a dot, it should be ".intList".
Fix proposals:
- Update linker script to contain a do in intList section name
- Update the python script to accept ".intList" or "intList" section names.

Reply:
We have to go with the second answer. The section names preceded with dots should not be used as user sections. They are not strictly forbidden but they should be avoided and they are making trouble for clang. So updating clang scripts it not an option.

Reply:
Done